### DISCUSSION TITLE
resumeState functionality

### DISCUSSION THREAD

**Discussion by hamidreza-hosseini** [seaweedfs/seaweedfs#2822] [Q&A] [answered]
Hi @chrislusf ,
I was reading weed master options to start and encounter with "-resumeState":
```
  -resumeState
    	resume previous state on start master server

```
What does "resumeState" do when we set it "true", could you explain it more?

**Comment by kmlebedev** [comment, marked as answer]
https://github.com/chrislusf/seaweedfs/issues/2804

This option prevents the snapshot from being removed from the raft state machine.
https://github.com/chrislusf/seaweedfs/blob/master/weed/server/raft_server.go#L86
